twos-complement/bigint2twos.go

```diff
@@ -0,0 +1,82 @@
+package twoscomplement
+
+import "math/big"
+
+var bigOne = big.NewInt(1)
+
+// ToBytes returns a variable length two's complement byte array representation of the input.
+// Does not alter input.
+func ToBytes(bi *big.Int) []byte {
+	var resultBytes []byte
+	switch bi.Sign() {
+	case -1:
+		// compute 2's complement
+		plus1 := big.NewInt(0)
+		plus1 = plus1.Add(bi, big.NewInt(1)) // add 1
+		resultBytes = plus1.Bytes()
+		for i, b := range resultBytes {
+			resultBytes[i] = ^b // negate every bit
+		}
+		if len(resultBytes) == 0 || resultBytes[0]>>7 != 1 {
```

Comment:
Also, replace `resultBytes[0] >> 7 != 1` with `resultBytes[0] >> 7 == 0`. Might want to extract it as a local variable `mostSignificantBit`.